beater/beater.go

```diff
@@ -192,7 +194,15 @@ func (bt *beater) Run(b *beat.Beat) error {
 		return nil
 	}
 
-	bt.server = newServer(bt.config, tracer, pub.Send)
+	var kb *kibana.Client
+	if bt.config.Kibana.Enabled() {
+		kb, err = kibana.NewKibanaClient(bt.config.Kibana)
+		if err != nil {
+			bt.logger.Error(err.Error())
```

Comment:
Why not return the error here?

With the current logic, `kb` should be set to nil if `err != nil`, to ensure you aren't passing a faulty or half initialized Kibana client on to `newServer`.

Reply:
An error will also be returned if config is valid but Kibana is not reachable.
Are you proposing to stop the server, then? I can do that. Although that'd diverge from output behaviour.

Reply:
ok, that's a good point. Is there any possibility to distinguish between those errors? I agree we should not stop the server when Kibana is not reachable.

Reply:
apm-server does never stop when any other config is malformed/faulty, right? (elasticsearch host, max_bulk_size, etc) 
if we want the kibana setting be an exception, i guess we can add support in libbeat for that

Reply:
Right, fine as is then.